src/EventSource.js

```diff
@@ -77,8 +77,6 @@ class EventSource {
       }
 
       this._xhr.setRequestHeader('Accept', 'text/event-stream');
-      this._xhr.setRequestHeader('Cache-Control', 'no-cache');
```

Comment:
The idea is good, however, these headers have been in place since the library was created and are the default behavior. Removing them may cause issues for users currently using react-native-sse. I suggest defining these headers as default and allowing users to `null` them, which will simply result in not setting them using `setRequestHeader`.

Reply:
I tried overwriting these headers. When I tried doing so, it is not overriding but appending the values.

For example: On adding `cache-control: no-store`, it makes appends and makes it` cache-control: no-cache, no-store`

I am not sure if it would cause issues for users currently using the library. If you think so, can you please provide an example? 

In my case, these 2 headers are not allowed in `Access-Control-Allow-Headers` of preflight response, hence creating CORS error.

Reply:
I think the cleanest solution would be to remove these two lines and leave it to the user to set those headers if they need them. Of course, this could theoretically lead to problems for exisiting users if they
1) upgrade to the new version of react-native-sse where the headers are removed,
2) have a setup that really requires those headers to be there, and
3) forget to set them manually after upgrading.

However, I think the number of users who check all three of the above boxes will be very small. To mitigate the problem, you could just mark this as a breaking change and put a warning in the release notes.

Reply:
@arpitmalik It's not about overwriting these values, as that won't work. The idea is to predefine these entries as `this.headers` and then merge them with `options.headers`. When setting headers using `setRequestHeader`, ensure to exclude those which have a value of "null" or more precisely, a value of "undefined".

react-native-sse is downloaded from npm approximately 20,000 times per week. I aim to avoid making modifications that might lead developers to waste time searching for a solution to the problem unknowingly.

Reply:
I have the same problem, cache-control headers is not allowed and is causing cors error, but when I set `'Cache-Control': undefined`, I get `'Cache-Control': no-cache, undefined`, so how can I unset this header?

Reply:
for those who have similar cases, my temp solution is: 
I locked the specific react-native-sse version, and use https://github.com/ds300/patch-package to creat a patch to remove these two lines

but this is not considered as a good idea though, and these custom headers are unexpected since default headers is `{}` as documented, @wojciechkrol  maybe you can add an option controlling whether to omit these headers? I am not sure if this is a proper suggestion, anyway, thanks for the efforts on the library, it helped a lot 🙏🙏

Reply:
If OP does not implement these changes, I plan to add this capability.

Reply:
I am on it ✅

Reply:
> I am on it ✅, Might be a foolish question but what do you mean by OP?

Original Poster 👀